### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by kunmi02**
This pull request contains the logic of the game.
The project was built using the object-oriented paradigm. Every method, conditions and loops used in this project are well planned out and thought of.

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      [@board[0][0], @board[1][1], @board[2][2]],
+      [@board[2][0], @board[1][1], @board[0][2]],
+      [@board[0][0], @board[1][0], @board[2][0]],
+      [@board[0][1], @board[1][1], @board[2][1]],
+      [@board[0][2], @board[1][2], @board[2][2]]
+
+    ]
+    combinations.any? do |combo|
+      combo.all? { |i| i == 'x' } || combo.all? { |i| i == 'o' }
+      # return true if combo.all? || combo.all?
```
Comment (elmejdki) [inline lib/game.rb]: remove this line, no commented code is allowed.

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# player
+class Player
+  attr_accessor :player_name, :symbol
```
Comment (elmejdki) [inline lib/player.rb]: change this to an attr_readers cause you don't need to reassign these values.

File: bin/main [hunk 3]
```diff
@@ -1,168 +1,155 @@
...
-def validate_name(name)
-  while name.strip == ''
-    p 'please add a valid name'
-    name = gets.chomp
+#!/usr/bin/ruby
+# rubocop:disable Metrics/BlockNesting, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+require_relative '../lib/player'
+require_relative '../lib/game'
+class UserInterface
+  require_relative '../lib/game'
```
Comment (elmejdki) [inline bin/main:6]: remove this extra require_relative

File: bin/main [hunk 4]
```diff
@@ -1,168 +1,155 @@
...
-
-def validate_name(name)
-  while name.strip == ''
-    p 'please add a valid name'
-    name = gets.chomp
+#!/usr/bin/ruby
+# rubocop:disable Metrics/BlockNesting, Metrics/MethodLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+require_relative '../lib/player'
+require_relative '../lib/game'
+class UserInterface
```
Comment (elmejdki) [inline bin/main:5]: add an empty line before the class

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## Nice work so far guy, and great job with having a decent Readme file :clap: :clap: :clap: :clap:, you still have some minor changes :hammer: :wrench: to take care of though.
## check the comments below :point_down: :point_down: :point_down: :point_down:

- [x] please remove the .gitkeep file from the lib directory.

- [x] your main file is not executable change the top line to this to fix it: `#!/usr/bin/env ruby`.

- [x] your board should be displayed correctly with or without empty cells.
Here we have a winner but we can figure it out from the board cause it's hard to read:
image

### Please make all the required changes before you submit another code review.
## Happy Coding :tophat:

**Comment by kunmi02** [comment]
> # status: require changes ♻️
> ## Nice work so far guy, and great job with having a decent Readme file 👏 👏 👏 👏, you still have some minor changes 🔨 🔧 to take care of though.
> ## check the comments below 👇 👇 👇 👇
> * [x]  please remove the .gitkeep file from the lib directory.
> * [x]  your main file is not executable change the top line to this to fix it: `#!/usr/bin/env ruby`.
> * [x]  your board should be displayed correctly with or without empty cells.
>   Here we have a winner but we can figure it out from the board cause it's hard to read:
>   image
> 
> ### Please make all the required changes before you submit another code review.
> ## Happy Coding 🎩

The main file is executable. `#!/usr/bin/env ruby`. exists in my main file. Please check again

File: bin/main [hunk 5]
```diff
@@ -1,168 +1,154 @@
-#!/usr/bin/env ruby
-# rubocop:disable Metrics/BlockNesting
```
Comment (SpaYco) [inline bin/main]: - [ ] please revert this to `#!/usr/bin/env ruby` to make the file executable ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/execute_bin.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] check below 👇 

## Optional 🟢 

- [ ] move all boards methods to a board class

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by krys2fa — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on making all the requested changes and getting this milestone approved. 👏  

🔆  You can now merge this branch into the development branch and go on with the next milestone. Ensure to submit a project completion form after merging this branch.

All the best team,
Chris.

**Pull request merged**